lib/certUtils.js

```diff
@@ -38,18 +38,10 @@ class Certificate {
 
 	static #searchForCommonName(attributes) {
 		const X509_COMMON_NAME_KEY = "2.5.4.3";
-
-		let commonName = ""; // Default in case no subject is found
-
 		// Search the attributes for the common name of the certificate
-		for (let index = 0; index < attributes.length; index++) {
-			const attribute = attributes[index];
-			if (attribute.type === X509_COMMON_NAME_KEY) {
-				commonName = attribute.value.valueBlock.value;
-				break;
-			}
-		}
-		return commonName;
+		const commonNameAttr = attributes.find((attr) => attr.type == X509_COMMON_NAME_KEY);
```

Comment:
I'm not sure where, but I recently read an article about certificate validation, that abused a bug due to different validations in the certificate chain. I can't make a 1:1 connection right now, but I think the security vulnerability was created by the program using an attribute down the chain, while browsers etc only validated the first one.

It would be great if we could stick with a for-loop, and just change the for-index to a for-of one.

Reply:
I had a hard look, and I think it was this one (focus on chapter 1): https://github.blog/2023-08-17-mtls-when-certificate-authentication-is-done-wrong/